Classes/Controller/BackendServiceController.php

```diff
@@ -327,6 +333,19 @@ public function changeBaseWorkspaceAction(string $targetWorkspaceName, NodeInter
         $this->view->assign('value', $this->feedbackCollection);
     }
 
+    /**
+     * Persists the clipboard state
+     *
+     * @param string $clipboardSubject
+     * @param string $clipboardMode
+     * @return void
+     */
+    public function persistClipboardStateAction(string $clipboardSubject, string $clipboardMode)
```

Comment:
What are `$clipboardSubject` and `$clipboardMode`?

Reply:
After skimming over the JS code below I think `clipboardSubject ` refers to the `contextPath` of a single node in the clipboard, and `clipboardMode` is either the string `'COPY'` or `'CUT'`, correct?

Why not sth like
```php
class ClipboardState {
  public function copyNode(NodeInterface $node): void {}

  public function cutNode(NodeInterface $node): void {}

  public function pasteNode(): NodeInterface {}
}
```

Reply:
Sure, why not. Will take care.

Reply:
Only if you agree!?

Reply:
Hah, well, I think it's more about the preference on which spectrum of abstraction one likes to be. E.g. due to my temperament I prefer to start with the least amount of abstractions and write the imperative code without any abstractions, and then refactor to add more abstractions when I got the gut feeling telling me to do so.
I know that you come from the DDD background of architecting large domain-centric systems that are there to stay and evolve, so I understand where your suggestions are coming from, and I'm ready to adapt to this style of coding and learn a few things along the way.